src/root/views/Country/ThreeW/common.ts

```diff
@@ -116,34 +142,75 @@ export function projectListToSankeyData(projectList: Project[]) {
     });
   });
 
-  const reducedLinks = links.reduce((acc, val) => {
-    const i = acc.findIndex(
-      l => l.source === val.source
-      && l.target === val.target
-    );
+  const reducedLinks = reduceLinks(links);
 
-    if (i === -1) {
-      return [...acc, val];
-    }
+  return {
+    nodes,
+    links: reducedLinks,
+  } as const;
+}
 
-    const newAcc = [...acc];
-    const {0: prevLink} = newAcc.splice(i, 1);
+export function projectListToNsSankeyData(projectList: Project[]) {
+  const nodes: SankeyNode[] = [];
+  const links: SankeyLink[] = [];
 
-    return [
-      ...newAcc, {
-        ...prevLink,
-        value: prevLink.value + 1
-      }
-    ];
-  }, [] as SankeyLink[]);
+  if (projectList.length === 0) {
+    return {
+      nodes,
+      links,
+    } as const;
+  }
+
+  nodes.push({ name: projectList[0].reporting_ns_detail.society_name });
+
+  const countryList = uniqueAndTransform(
+    projectList,
+    d => d.project_country,
+    d => d.project_country_detail,
+  );
+  const countryIdToNodeIndexMap: Record<string, number> = {};
+  countryList.forEach((country) => {
+    countryIdToNodeIndexMap[country.id] = nodes.push({
+      name: country.name,
+    }) - 1;
+  });
+
+  const primarySectorList = uniqueAndTransform(
+    projectList,
+    d => d.primary_sector,
+    d => ({ id: d.primary_sector, label: d.primary_sector_display }),
+  );
+  const primarySectorIdToNodeIndexMap: Record<string, number> = {};
+  primarySectorList.forEach((ps) => {
+    primarySectorIdToNodeIndexMap[ps.id] = nodes.push({
+      name: ps.label,
+    }) - 1;
+  });
```

Comment:
This looks a bit weird.
Plus, why not use listToMap?

Reply:
It is weird

Reply:
Can't we simplify this?

Reply:
Lets try